pkg/util/types/patch.go

```diff
@@ -19,8 +19,55 @@
 
 package types
 
+import (
+	"encoding/json"
+	"fmt"
+)
+
 type PatchOperation struct {
 	Op    string      `json:"op"`
 	Path  string      `json:"path"`
 	Value interface{} `json:"value,omitempty"`
 }
+
+const (
+	PatchReplaceOp = "replace"
+	PatchTestOp    = "test"
+	PatchAddOp     = "add"
+	PatchRemoveOp  = "remove"
+)
+
+func GeneratePatchPayload(patches ...PatchOperation) ([]byte, error) {
+	if len(patches) == 0 {
+		return nil, fmt.Errorf("list of patches is empty")
+	}
+
+	payloadBytes, err := json.Marshal(patches)
+	if err != nil {
+		return nil, err
+	}
+
+	return payloadBytes, nil
+}
+
+func GenerateTestReplacePatch(path string, oldValue, newValue interface{}) ([]byte, error) {
+	return GeneratePatchPayload(
+		PatchOperation{
+			Op:    PatchTestOp,
+			Path:  path,
+			Value: oldValue,
+		},
+		PatchOperation{
+			Op:    PatchReplaceOp,
+			Path:  path,
+			Value: newValue,
+		},
+	)
+}
+
+func UnmarshalPatch(patch []byte) ([]PatchOperation, error) {
+	var p []PatchOperation
+	err := json.Unmarshal(patch, &p)
+
```

Comment:
```suggestion
func UnmarshalPatch(patch []byte) (p []PatchOperation, err error) {
	err = json.Unmarshal(patch, &p)
```

Reply:
I think named return variables should be avoided, since they make code less explicit.

The advantage you get is one less line of code - I don't think that has enough value.

Reply:
My opinion is sort of the middle ground between the two opinions.

My view on this is that named return values are valuable and should be used in some situations. These might be:
- The less important one: to save lines of code that only initialize new variables. These are absolutely redundant and doesn't make the code clearer (or even makes it worse).
- The more important one: to "document" the return value if its purpose is not clear from the type only. It's much better to document it with a variable name than with code comments IMO.

In this situation only one variable needs to be explicitly initialized, but on the other hand both of the return values purpose is very clear (that's why their names don't provide any information / documentation). So it's kind of the borderline IMO.

I think it can stay as it is.